File: app/src/main/java/com/gelostech/pocketbartender/fragments/FavesFragment.java
package com.gelostech.pocketbartender.fragments;


import android.content.Context;
import android.content.Intent;
import android.graphics.Canvas;
import android.graphics.Typeface;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import com.gelostech.pocketbartender.R;
import com.gelostech.pocketbartender.activities.CocktailActivity;
import com.gelostech.pocketbartender.adapters.FavesAdapter;
import com.gelostech.pocketbartender.commoners.DatabaseHelper;
import com.gelostech.pocketbartender.models.FavesModel;
import com.gelostech.pocketbartender.models.HomeModel;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * A simple {@link Fragment} subclass.
 */
public class FavesFragment extends Fragment {
    private View view;
    private final String font1 = "fonts/COCKB___.TTF";
    private final String font2 = "fonts/COCKTAIL.TTF";
    private FavesAdapter favesAdapter;
    private List<HomeModel> faves;
    private DatabaseHelper db;

    @BindView(R.id.no_faves_tv) TextView noFaves;
    @BindView(R.id.faves_toolbar) Toolbar toolbar;
    @BindView(R.id.faves_title) TextView favesTitle;
    @BindView(R.id.faves_rv) RecyclerView rv;
    @BindView(R.id.faves_loading) ProgressBar progressBar;

    public FavesFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
         if(view == null){
             view = inflater.inflate(R.layout.fragment_faves, container, false);
             ButterKnife.bind(this, view);

             db = new DatabaseHelper(getActivity());
             initViews();
             loadFaves();
         }

         return view;
    }

    private void initViews(){
        ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
        ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(null);
        favesTitle.setTypeface(Typeface.createFromAsset(getActivity().getAssets(), font1));
        
        rv.setHasFixedSize(true);
        rv.setLayoutManager(new LinearLayoutManager(getContext()));
        rv.addItemDecoration(new SimpleDividerItemDecoration(getActivity()));

        faves = new ArrayList<>();

    }

    private void loadFaves(){
        faves = db.fetchFaves(getActivity());

        if(faves.size() == 0){
            progressBar.setVisibility(View.GONE);
            noFaves.setVisibility(View.VISIBLE);
        } else {
            favesAdapter = new FavesAdapter(getActivity(), faves, new FavesAdapter.OnItemClickListener() {
                @Override
                public void onItemClick(HomeModel model, int viewID, int position) {
                    switch (viewID){
                        case 0:
                            Toast.makeText(getActivity(), "liked", Toast.LENGTH_SHORT).show();
                            break;

                        case 1:
                            model.setCocktailThumb(null);

                            Intent intent = new Intent(getActivity(), CocktailActivity.class);
                            intent.putExtra("cocktail", model);
                            startActivity(intent);
                            getActivity().overridePendingTransition(R.anim.enter_b, R.anim.exit_a);
                            break;

                        default:
                            break;
                    }
                }
            });
            rv.setAdapter(favesAdapter);

            rv.setVisibility(View.VISIBLE);
            progressBar.setVisibility(View.GONE);

        }


    }

    public class SimpleDividerItemDecoration extends RecyclerView.ItemDecoration {
        private Drawable mDivider;

        public SimpleDividerItemDecoration(Context context) {
            mDivider = ContextCompat.getDrawable(context,R.drawable.recycler_divider);
        }

        @Override
        public void onDrawOver(Canvas c, RecyclerView parent, RecyclerView.State state) {
            int left = parent.getPaddingLeft();
            int right = parent.getWidth() - parent.getPaddingRight();

            int childCount = parent.getChildCount();
            for (int i = 0; i < childCount; i++) {
                View child = parent.getChildAt(i);

                RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child.getLayoutParams();

                int top = child.getBottom() + params.bottomMargin;
                int bottom = top + mDivider.getIntrinsicHeight();

                mDivider.setBounds(left, top, right, bottom);
                mDivider.draw(c);
            }
        }
    }


}

File: app/src/main/java/com/gelostech/pocketbartender/fragments/FavesFragment.kt
package com.gelostech.pocketbartender.fragments

import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.gelostech.pocketbartender.R
import com.gelostech.pocketbartender.activities.CocktailActivity
import com.gelostech.pocketbartender.activities.MainActivity
import com.gelostech.pocketbartender.adapters.FavesAdapter
import com.gelostech.pocketbartender.commoners.BartenderUtil
import com.gelostech.pocketbartender.commoners.DatabaseHelper
import com.gelostech.pocketbartender.models.HomeModel
import com.gelostech.pocketbartender.utils.RecyclerFormatter
import com.gelostech.pocketbartender.utils.setFont
import kotlinx.android.synthetic.main.fragment_faves.*
import kotlinx.android.synthetic.main.fragment_faves.view.*
import com.gelostech.pocketbartender.utils.snack
import org.jetbrains.anko.alert
import org.jetbrains.anko.toast
import org.jetbrains.anko.yesButton

/**
 * A simple [Fragment] subclass.
 */
class FavesFragment : Fragment(), FavesAdapter.OnItemClickListener {
    private var favesAdapter: FavesAdapter? = null
    private var faves: MutableList<HomeModel>? = null
    private var db: DatabaseHelper? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_faves, container, false)
        setHasOptionsMenu(false)

        db = DatabaseHelper(activity)
        initViews(view)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        loadFaves()
        super.onViewCreated(view, savedInstanceState)
    }

    private fun initViews(v: View) {
        setHasOptionsMenu(true)
        (activity as MainActivity).setSupportActionBar(v.favesToolbar)
        (activity as MainActivity).supportActionBar?.title = null
        v.favesToolbarTitle.setFont(BartenderUtil.FONT)
        v.favesToolbarTitle.text = getString(R.string.faves)

        v.favesRv.setHasFixedSize(true)
        v.favesRv.layoutManager = LinearLayoutManager(context)
        v.favesRv.addItemDecoration(RecyclerFormatter.SimpleDividerItemDecoration(activity!!))

        faves = mutableListOf()

    }

    private fun loadFaves() {
        if (faves!!.size > 0) faves!!.clear()
        faves = db!!.fetchFaves()

        if (faves!!.size == 0) {
            view!!.favesLoading.visibility = View.GONE
            view!!.favesNoContent.visibility = View.VISIBLE
            if (view!!.favesRv.isShown) favesRv!!.visibility = View.GONE

        } else {
            view!!.favesNoContent?.visibility = View.GONE

            favesAdapter = FavesAdapter(activity, faves, this)
            view!!.favesRv.adapter = favesAdapter
            view!!.favesRv.visibility = View.VISIBLE
            view!!.favesLoading.visibility = View.GONE

        }

    }

    override fun onItemClick(model: HomeModel, viewID: Int, position: Int) {
        when (viewID) {
            0 -> {
                activity!!.alert("Remove ${model.name} from favorites?") {
                    yesButton {
                        db!!.deleteRow(model.id.toString(), DatabaseHelper.FAVES_TABLE)
                        BartenderUtil.removeFaveCocktail(activity!!, model.id.toString())
                        view?.snack("${model.name} removed from favorites"){}
                        favesAdapter!!.removeFave(position)

                        if(faves!!.size == 0) {
                            view!!.favesNoContent.visibility = View.VISIBLE
                            if (view!!.favesRv.isShown) favesRv!!.visibility = View.GONE
                        }
                    }
                }.show()
            }

            1 -> {
                model.cocktailThumb = null

                val intent = Intent(activity, CocktailActivity::class.java)
                intent.putExtra("cocktail", model)
                startActivity(intent)
                activity!!.overridePendingTransition(R.anim.enter_b, R.anim.exit_a)
            }

            else -> {
            }
        }
    }

    override fun onResume() {
        loadFaves()
        super.onResume()
    }
}
